File: engine-src/packages/parsers/python/test/test_ir_basic.js
// 單元測試：Python IR 分析（Node.js 版）
const { parsePythonProject } = require('../dist/index');

function test_simple_class() {
    const files = {
        'mod1.py': `
class A:
    def foo(self):
        pass
class B(A):
    def bar(self):
        self.foo()
`
    };
    const ir = parsePythonProject(files);
    if (!ir.modules['mod1']) throw new Error('mod1 not found');
    const mod = ir.modules['mod1'];
    if (!mod.classes.some(c => c.name === 'A')) throw new Error('Class A not found');
    if (!mod.classes.some(c => c.name === 'B')) throw new Error('Class B not found');
    if (!mod.functions.concat(mod.classes[0].methods).some(f => f.name === 'foo')) throw new Error('foo not found');
    if (!mod.functions.concat(mod.classes[1].methods).some(f => f.name === 'bar')) throw new Error('bar not found');
    if (!mod.classes[1].bases.includes('A')) throw new Error('B does not inherit A');
    if (!mod.classes[1].methods[0].calls.includes('self.foo')) throw new Error('bar does not call self.foo');
}

function test_simple_func() {
    const files = {
        'mod2.py': `
def hello():
    print("hi")
`
    };
    const ir = parsePythonProject(files);
    if (!ir.modules['mod2']) throw new Error('mod2 not found');
    const mod = ir.modules['mod2'];
    if (!mod.functions.some(f => f.name === 'hello')) throw new Error('hello not found');
    if (!mod.functions[0].calls.includes('print')) throw new Error('hello does not call print');
}

function runAll() {
    test_simple_class();
    test_simple_func();
    console.log('All Python IR tests passed.');
}

runAll();

File: engine-src/packages/parsers/python/test/test_ir_basic.py
"""
Unit Test: Python IR Analysis (Basic Classes/Functions/Calls/Inheritance)
"""
import sys
sys.path.insert(0, '../../src')
from index import parsePythonProject

def test_simple_class():
    files = {
        'mod1.py': '''
class A:
    def foo(self):
        pass
class B(A):
    def bar(self):
        self.foo()
'''
    }
    ir = parsePythonProject(files)
    assert 'mod1' in ir['modules']
    mod = ir['modules']['mod1']
    assert any(c['name'] == 'A' for c in mod['classes'])
    assert any(c['name'] == 'B' for c in mod['classes'])
    assert any(f['name'] == 'foo' for f in mod['functions'] + mod['classes'][0]['methods'])
    assert any(f['name'] == 'bar' for f in mod['functions'] + mod['classes'][1]['methods'])
    # 檢查繼承
    assert 'A' in mod['classes'][1]['bases']
    # 檢查呼叫
    assert 'self.foo' in mod['classes'][1]['methods'][0]['calls']

def test_simple_func():
    files = {
        'mod2.py': '''
def hello():
    print("hi")
'''
    }
    ir = parsePythonProject(files)
    assert 'mod2' in ir['modules']
    mod = ir['modules']['mod2']
    assert any(f['name'] == 'hello' for f in mod['functions'])
    assert 'print' in mod['functions'][0]['calls']

if __name__ == '__main__':
    test_simple_class()
    test_simple_func()
    print('All Python IR tests passed.')
